File: 3483-unique-3-digit-even-numbers/solution.py
from collections import Counter

class Solution:
    def countEvenNumbers(self, digits: list[int]) -> int:
        # Step 1: Count available digit frequencies
        freq = Counter(digits)
        distinct_count = 0
        
        # Step 2: Iterate through all valid 3-digit even numbers
        for num in range(100, 1000, 2):
            hundred = num // 100
            ten = (num // 10) % 10
            unit = num % 10
            
            num_freq = Counter([hundred, ten, unit])
            
            # Step 3: Check if available digits can construct this number
            if all(num_freq[d] <= freq[d] for d in num_freq):
                distinct_count += 1
                
        return distinct_count

File: 3483-unique-3-digit-even-numbers/solution.cpp
#include <vector>

class Solution {
public:
    int countEvenNumbers(std::vector<int>& digits) {
        // Step 1: Count frequency of available digits
        std::vector<int> freq(10, 0);
        for (int digit : digits) {
            freq[digit]++;
        }
        
        int distinctCount = 0;
        
        // Step 2: Check every 3-digit even number from 100 to 998
        for (int num = 100; num <= 998; num += 2) {
            int hundred = num / 100;
            int ten = (num / 10) % 10;
            int unit = num % 10;
            
            std::vector<int> currentFreq(10, 0);
            currentFreq[hundred]++;
            currentFreq[ten]++;
            currentFreq[unit]++;
            
            // Step 3: Verify if current number can be formed
            bool isValid = true;
            for (int d = 0; d < 10; ++d) {
                if (currentFreq[d] > freq[d]) {
                    isValid = false;
                    break;
                }
            }
            
            if (isValid) {
                distinctCount++;
            }
        }
        
        return distinctCount;
    }
};